libqtile/state.py

```diff
@@ -45,6 +53,39 @@ def apply(self, qtile):
             Rearrange the windows in the specified Qtile object according to
             this QtileState.
         """
+        try:
+            for group in qtile.groups:
+                x = []
+                for i in self.focusHistory[group.name]:
+                    x.append(qtile.windowMap[i.Wid])
+                group.focusHistory = x
+                for layout in group.layouts:
```

Comment:
`__getstate__` & `__setstate__` methods made it possible to pickle the layout class and window class. The group class is still *unpickeble* because of recursive attributes.(`group` has a link to `qtile` which has a link back to `group`). 
We don't need to pickle the group class , only one of it attributes `focusHistory` was enough for everything to get working and so I directly pickled it.If we restore the group state back to original , any changes made in configuration would not reflect after restart. Hence I only pickled it's attribute `focusHistory`.